src/Future.js

```diff
@@ -82,6 +84,79 @@ Future.prototype.chain = function(f) {  // Sorella's:
   }.bind(this));
 };
 
+var chainRecFork = function(t, rej, res) {
+  var isSync = false;
+  t.fork(function(v) {
+    var r = rej(v, !isSync);
+    isSync = true;
+    return r;
+  }, function(v) {
+    var r = res(v, !isSync);
+    isSync = true;
+    return r;
+  });
+  if (isSync) {
+    return { isSync: true };
+  } else {
+    isSync = true;
+    return { isSync: false };
+  }
+};
+
+// chainRec
+Future.chainRec = Future.prototype.chainRec = function(f, i) {
```

Comment:
I have only looked at [implementation for Eff](https://github.com/purescript/purescript-tailrec/blob/master/src/Control/Monad/Rec/Class.purs#L97-L112) but I don't use and know purescript and it was too criptic for me.

The implementation you have linked and ported is nicer! Will update it 🌴